metadata.go

```diff
@@ -192,6 +200,86 @@ func GetImageToMDMap(imageMDs []ImageMetadataInfo) (imageToMDMap map[string][]Im
 	return
 }
 
+// getLocalImages queries the local Docker daemon for list of images.
+func getLocalImages() (imageMap map[ImageIDType][]RepoTagType, e error) {
+
+	// query a list of images from Docker daemon
+	response, e := listImages()
+	if e != nil {
+		return nil, e
+	}
+	blog.Info(string(response))
+	// parse JSON
+	var localImageList []LocalImageStruct
+	if e = json.Unmarshal(response, &localImageList); e != nil {
+		return nil, e
+	}
+
+	// make map from each imageID to all of its aliases (repo+tag)
+	imageMap = make(map[ImageIDType][]RepoTagType)
+	for _, localImage := range localImageList {
+		imageID := ImageIDType(localImage.ID)
+		for _, repoTag := range localImage.RepoTags {
+			// skip images with no repo:tag
+			if repoTag == "" {
+				blog.Info("Image ", imageID, " has a repoTag of empty string.")
+				continue
+			}
+			// repoTag example: "localhost:5000/test/busybox:latest"
+			// repo: "localhost:5000/test/busybox"
+			// tag: "latest"
+			ss := strings.Split(repoTag, ":")
+			if len(ss) < 2 {
+				blog.Info("Image ", imageID, " has a wrong repoTag with a wrong format, ", repoTag)
+				continue
+			}
+			tag := ss[len(ss)-1]
+			repo := repoTag[:len(repoTag)-len(tag)-1]
+			blog.Debug(imageID, repoTag, repo, tag)
+
+			// If the map ReposToProcess is not empty, skip images that are not from the the repos in the map
+			if len(ReposToProcess) > 0 && !ReposToProcess[RepoType(repo)] {
+				continue
+			}
+			repotag := RepoTagType{Repo: RepoType(repo), Tag: TagType(tag)}
+
+			if _, ok := imageMap[imageID]; ok {
```

Comment:
I think it would be cleaner to declare a named type for this map, and then define an Insert() method.